eth_keys/main.py

```diff
@@ -21,57 +14,17 @@
 )
 
 
-def backend_property_proxy(name):
-    @property
-    def property_fn(self):
-        backend_property = getattr(self.backend, name)
-        return backend_property
-    return property_fn
-
-
-class KeyAPI(object):
-    backend = None
-
-    def __init__(self, backend=None):
-        if backend is None:
-            pass
-        elif isinstance(backend, BaseECCBackend):
-            pass
-        elif isinstance(backend, type) and issubclass(backend, BaseECCBackend):
-            backend = backend()
-        elif is_string(backend):
-            backend = get_backend(backend)
-        else:
-            raise ValueError(
-                "Unsupported format for ECC backend.  Must be an instance or "
-                "subclass of `eth_keys.backends.BaseECCBackend` or a string of "
-                "the dot-separated import path for the desired backend class"
-            )
-
-        self.backend = backend
-
-    _backend = None
-
-    @property
-    def backend(self):
-        if self._backend is None:
-            return get_backend()
-        else:
-            return self._backend
-
-    @backend.setter
-    def backend(self, value):
-        self._backend = value
+class KeyAPI(LazyBackend):
+    #
+    # datatype shortcuts
+    #
+    PublicKey = PublicKey  # type: Any
+    PrivateKey = PrivateKey  # type: Any
+    Signature = Signature  # type: Any
```

Comment:
Since they're no longer dynamically computed, can we use specific types here?

Reply:
This is me getting confused with futzing with mypy.  Yes we should be able to.   Here is what I'm currently getting when I specify these as their types.

```bash
$ mypy --ignore-missing-imports -p eth_keys
eth_keys/main.py:21: error: Invalid type "eth_keys.main.KeyAPI.PublicKey"
eth_keys/main.py:22: error: Invalid type "eth_keys.main.KeyAPI.PrivateKey"
eth_keys/main.py:23: error: Invalid type "eth_keys.main.KeyAPI.Signature"
eth_keys/main.py:28: error: Invalid type "eth_keys.main.KeyAPI.Signature"
eth_keys/main.py:30: error: Invalid type "eth_keys.main.KeyAPI.PrivateKey"
eth_keys/main.py:48: error: Invalid type "eth_keys.main.KeyAPI.Signature"
eth_keys/main.py:49: error: Invalid type "eth_keys.main.KeyAPI.PublicKey"
eth_keys/main.py:58: error: Invalid type "eth_keys.main.KeyAPI.PublicKey"
eth_keys/main.py:60: error: Invalid type "eth_keys.main.KeyAPI.Signature"
/Users/piper/sites/eth-keys/eth_keys/main.py:34: error: INTERNAL ERROR -- please report a bug at https://github.com/python/mypy/issues version: 0.560
/Users/piper/sites/eth-keys/eth_keys/main.py:34: : note: please use --show-traceback to print a traceback when reporting a bug
```

Reply:
Hmm, looks like mypy is resolving the names in your type annotations to the attributes of the KeyAPI class with the same names, and those are indeed not types. Should probably file a bug against mypy, but in the meantime maybe it'd work if you used `PublicKey = PublicKey  # type: Type[datatypes.PublicKey]`?

Reply:
It has already been reported: https://github.com/python/mypy/issues/3775